forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -1293,33 +1293,18 @@ public final void checkStateEffects(final boolean runEvents, final Set<Card> aff
                         noRegCreats.add(c);
                         checkAgain = true;
                     } else if (c.hasKeyword("CARDNAME can't be destroyed by lethal damage unless lethal damage dealt by a single source is marked on it.")) {
-                        // merge entries with same source
-                        List<Integer> dmgList = Lists.newArrayList();
-                        List<Pair<Card, Integer>> remainingDamaged = Lists.newArrayList(c.getReceivedDamageFromThisTurn());
-                        while (!remainingDamaged.isEmpty()) {
-                            Pair <Card, Integer> damaged = remainingDamaged.get(0);
-                            int sum = damaged.getRight();
-                            remainingDamaged.remove(damaged);
-                            for (Pair<Card, Integer> other : Lists.newArrayList(remainingDamaged)) {
-                                if (other.getLeft().equalsWithTimestamp(damaged.getLeft())) {
-                                    sum += other.getRight();
-                                    // once it got counted keep it out
-                                    remainingDamaged.remove(other);
-                                }
-                            }
-                            dmgList.add(sum);
+                        int maxDmg = 0;
+                        for (Pair<GameEntity, Integer> p : game.getDamageDoneThisTurn(null, false, false, null, "Card.StrictlySelf", c, null, null)) {
```

Comment:
i still would have kept this info still on the card, so it is always cleaned when it left the zone

so no stuff with StictlySelf needed

Reply:
the stuff that checks for stictlySelf should probably be still stored in Card(DamageHistory)

Reply:
Hmn, my idea for this PR was to deduplicate most of this stuff in a single place

But I think I can still make that part cleaner and allow Cards to be compared directly for these cases